### PR TITLE
Totem

### PR DESCRIPTION
**Pull request by Tyrannicodin**

### CODE REVIEW

File: common/cards/card-plugins/effects/totem.js [hunk 1]
```diff
@@ -9,31 +8,37 @@ class TotemEffectCard extends EffectCard {
...
-		// attacks
-		game.hooks.attack.tap(this.id, (target) => {
-			if (target.effectCardId === this.id) {
-				target.recovery.push({amount: this.recoverAmount, discardEffect: true})
-			}
-			return target
-		})
-
+	onAttach(game, instance, pos) {
 		game.hooks.hermitDeath.tap(this.id, (recovery, deathInfo) => {
```
Comment (NikoUY) [inline common/cards/card-plugins/effects/totem.js:22]: Use the afterAttack hook and this needs to be redone to work with that hook, make sure to discard the totem.

File: common/cards/card-plugins/effects/totem.js [hunk 2]
```diff
@@ -9,31 +8,37 @@ class TotemEffectCard extends EffectCard {
...
+					amount: 10,
+					discardEffect: true,
+				})
+			}
 			return recovery
 		})
 	}
+
+	/**
+	 *
```
Comment (NikoUY) [inline common/cards/card-plugins/effects/totem.js:34]: This can be removed

File: common/cards/card-plugins/effects/totem.js [hunk 3]
```diff
@@ -9,30 +8,35 @@ class TotemEffectCard extends EffectCard {
...
-		// attacks
-		game.hooks.attack.tap(this.id, (target) => {
-			if (target.effectCardId === this.id) {
-				target.recovery.push({amount: this.recoverAmount, discardEffect: true})
-			}
-			return target
-		})
+	onAttach(game, instance, pos) {
+		pos.otherPlayer.hooks.afterAttack[instance] = (result) => {
+			if (!result.attack.target.row || result.attack.target.row.health > 0)
```
Comment (Sense101) [inline common/cards/card-plugins/effects/totem.js:22]: Please check explicitly that the row health is not undefined. You can also define a "row" variable which is result.attack.target.row, to keep the code cleaner.

**Review by Sense101 — CHANGES_REQUESTED** [review]
One more change., then this can be merged :)

**Pull request merged**